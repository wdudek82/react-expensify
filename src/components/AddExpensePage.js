import React from 'react';

const AddExpensePage = () => (
  <div>
    This is Add Expense Page
  </div>
);

export default AddExpensePage;
