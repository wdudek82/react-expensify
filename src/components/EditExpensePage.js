import React from 'react';

const EditExpensePage = (props) => {
  console.log(props);
  const id = props.match.params.id;

  return (
    <div>
      <h2>Editing id: {id}</h2>
      Edit Expense Page
    </div>
  );
};

export default EditExpensePage;