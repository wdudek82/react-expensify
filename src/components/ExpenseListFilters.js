import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter } from '../actions/filters';

const handleInputChange = (e, dispatch) => {
  dispatch(
    setTextFilter(e.target.value)
  );
};

const ExpenseListFilters = (props) => (
  <div>
    <input
      type="text"
      value={props.filters.text}
      onChange={(e) => handleInputChange(e, props.dispatch)}
    />
  </div>
);

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

export default connect(mapStateToProps)(ExpenseListFilters);