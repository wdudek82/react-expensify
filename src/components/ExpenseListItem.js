import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <div key={id}>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>{amount} - created: {createdAt}</p>
  </div>
);

export default ExpenseListItem;