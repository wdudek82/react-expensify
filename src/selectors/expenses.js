const getVisibleExpenses = (
  expenses,
  { text, sortBy, startDate, endDate }
) => {
  return expenses.filter(expense => {
    const startDateMatch = typeof startDate !== 'number' || expense.createdAt >= startDate;
    const endDateMatch = typeof endDate !== 'number' || expense.createdAt <= endDate;
    
    const expenseDescription = expense.description.toLowerCase();
    const includesText = expenseDescription.includes(text.toLowerCase());

    return startDateMatch && endDateMatch && includesText;
  }).sort((a, b) => {
    if (sortBy === 'date') {
      return a.createdAt < b.createdAt;
    } else if (sortBy === 'amount') {
      return a.amount < b.amount;
    }
  });
};

export default getVisibleExpenses;