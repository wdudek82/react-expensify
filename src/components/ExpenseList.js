import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => {
  const expenses = (
    props.expenses.length === 0 ?
    <p>No expenses</p> :
    props.expenses.map((expense, index) => (
      <ExpenseListItem
        key={expense.id}
        {...expense}
      />
    )
  ));

  return (
    <div>
      {expenses}
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);

// const ConnectedExpenseList = connect((state) => {
//   return {
//     expenses: state.expenses
//   };
// })(ExpenseList);
// 
// export default ConnectedExpenseList;