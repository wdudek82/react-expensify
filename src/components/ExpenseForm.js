import React from 'react';
import { connect } from 'react-redux';
import { addExpense } from '../actions/expenses';


class ExpenseForm extends React.Component {
  state = {
    description: '',
    amount: 0,
    createdAt: 0,
    note: ''
  }

  handleSubmitForm = (e) => {
    e.preventDefault();
    this.props.dispatch(
      addExpense({ ...this.state })
    );
    this.setState(() => ({
      description: '',
      amount: 0,
      createdAt: 0,
      note: ''
    }));
  };

  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };

  onAmountChange = (e) => {
    const amount = parseInt(e.target.value) || 0;
    this.setState(() => ({ amount }));
  }

  onDateChange = (e) => {
    const createdAt = parseInt(e.target.value) || 0;
    this.setState(() => ({ createdAt }));
  }

  onNoteChange = (e) => {
    const note = e.target.value;
    this.setState(() => ({ note }));
  };

  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <form onSubmit={this.handleSubmitForm}>
          <div>
            <input
              type="text"
              placeholder="Description"
              autoFocus
              value={this.state.description}
              onChange={this.onDescriptionChange}
            />
          </div>
          <div>
            <input
              type="number"
              placeholder="Amount"
              value={this.state.amount}
              onChange={this.onAmountChange}
            />
          </div>
          <div>
            <input
              type="number"
              placeholder="Date"
              value={this.state.createdAt}
              onChange={this.onDateChange}
            />
          </div>
          <div>
            <textarea
              placeholder="Add a note for your expense (optional)"
              value={this.state.note}
              onChange={this.onNoteChange}
            ></textarea>
          </div>
          <input type="submit" value="Add Expense" />
        </form>
      </div>
    );
  }
}

export default connect()(ExpenseForm);
