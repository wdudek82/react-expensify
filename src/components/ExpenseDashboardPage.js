import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

class ExpenseDashboardPage extends React.Component {
  render() {
    return (
      <div>
        <ExpenseListFilters />
        <ExpenseList />
      </div>
    );
  };
}

export default ExpenseDashboardPage;