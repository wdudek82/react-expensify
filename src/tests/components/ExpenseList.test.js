import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import { ExpenseList } from '../../components/ExpenseList';

it('should render ExpenseList with expenses', () => {
  const wrapper = shallow(<ExpenseList expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});

it('should render expense list with empty message', () => {
  const wrapper = shallow(<ExpenseList expenses={[]} />);
  expect(wrapper).toMatchSnapshot();
})