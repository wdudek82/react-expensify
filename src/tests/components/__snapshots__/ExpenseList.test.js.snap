// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <ExpenseListItem
    amount={195}
    createdAt={0}
    description="Gum"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={100.95}
    createdAt={172800000}
    description="Water Bill"
    id="2"
    key="2"
    note="First note"
  />
  <ExpenseListItem
    amount={200.95}
    createdAt={-691200000}
    description="Gas  Bill"
    id="3"
    key="3"
    note="First note"
  />
  <ExpenseListItem
    amount={109500}
    createdAt={-345600000}
    description="Rent"
    id="4"
    key="4"
    note=""
  />
  <ExpenseListItem
    amount={4500}
    createdAt={345600000}
    description="Credit Card"
    id="5"
    key="5"
    note=""
  />
</div>
`;

exports[`should render expense list with empty message 1`] = `
<div>
  <p>
    No expenses
  </p>
</div>
`;
